app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -112,7 +112,7 @@ def build_select_button(bgi, index)
   def apply_common_props(button, input)
     button.update(
       :color        => input[:color],
-      :data         => input[:data],
+      :data         => input[:data] || (input[:klass].present? ? input[:klass].data(@record) : nil),
```

Comment:
I'd much rather see just `:data => button.data` here. The default implementation of the `data` method should take care of the logic.

(Also not sure why this is a static method + record argument instead of using an instance method?)

Reply:
Resolved